### ISSUE TITLE
Circularity between actOfPlanning and plannedAct

### ISSUE THREAD

**Issue by avsculley** [CommonCoreOntology/CommonCoreOntologies#176]
> cco:ActOfPlanning "A Planned Act that involves making a Plan to achieve some specified Objective."
> 
> cco:IntentionalAct "An Act in which at least one Agent plays a causative role and which is prescribed by some Directive Information Content Entity held by at least one of the Agents."

On the assumption IntentionalAct = PlannedAct (as seems the case in the new release), the DICE mentioned in the definition of IntentionalAct/PlannedAct is some Plan.

If so, then 'Act of planning' is a 'planned act' in a problematic way. Specifically, an act of planning requires a plan, and, of course, to get a plan you need some previous act of planning. Therefore, some act of planning requires some previous act of planning. But the previous act of planning itself requires a prior act of planning, and so on.

Put another way: definitions of processes should be success terms (they should specify the conditions under which it is true that a certain process has occurred). But for there to be a successful act of planning, there needs to be a successful planned act, and for that successful planned act, there needs to have been a successful act of planning, and so on.

**Comment by avsculley** [comment]
It requires a plan in that it is defined as a planned act.

**Comment by cameronmore** [comment]
> If so, then 'Act of planning' is a 'planned act' in a problematic way.

@avsculley I don't know that this is the case. A Plan is stronger, in the sense that it lays out more explicit steps and goals for an Agent than just a Directive ICE. When I sit down to formulate a plan, I already have a sense, an idea, of what a Plan should be--it should have achievable steps that follow each other in sequence which bring about some end state. None of that is required for a DICE.

It is not a trivial thing to learn how to plan, or how to plan well, and therefore I do not think there is a circularity

(open to being wrong of course)

**Comment by avsculley** [comment]
@cameronmore I'm not sure why I used the word 'circularity' originally, as it is just the wrong word to use. The idea is just that, since they are planned acts, all acts of planning are prescribed by a plan. But this can't be true becasue the first plan must be created by an act of planning that does not already depend on a plan prescribing it.

Your response seems to suggest that some other DICE, and intention, say, can give rise to a plan, which I completely agree with. The problem now is that either: (1) there are some acts of planning that are not planned acts, in the sense that they are not prescribed by a plan, in which case 'act of planning' should not be a subclass of 'planned act'; or (2) there are some planned acts that do not need to be prescribed by a plan, in which case the label 'planned act' needs to be changed becasue it, at best, is extremely misleading.

**Comment by cameronmore** [comment]
I agree for sure, this might be a problem of labeling. While the class "Planned Act" does have plan in the name, it is only defined in terms of Directive ICEs, so not cco:Plan. A more accurate label, according to the definition, would be Directed Acts (or the original label, Intentional Act). So, a Planned Act arises because of a DICE, and that Planned Act results in the creation of a Plan.

**Comment by avsculley** [comment]
The original label would also be wrong. Directed or Intended would make most sense. There are intentional acts that are not prescribed by DICEs whereas all intended acts are prescribed by DICEs. Plus, non-human agents can participate in intended acts, but not intentional acts.

**Comment by cameronmore** [comment]
Are you able to create a PR with the changes you have in mind to look at?

**Comment by avsculley** [comment]
Yes, I can do it this weekend.
________________________________

**Comment by neilotte** [comment]
@cameronmore @avsculley I'd recommend listing all the changes to be made on the thread before creating a PR. It looks like the PR that is open now only addresses some and not all of the changes to be made.

**Comment by cameronmore** [comment]
I think some consensus building / socializing the change would be required as well, but this initial commit from @avsculley clarifies what is being changed.

**Comment by johnbeve** [comment]
Two comments: 
- "Planned Act" as a parent of "Act of Planning" is confusing, so I'm in favor of adjusting relevant labels. 
- +1 to @neilotte's request for a list here of the changes to be made. @avsculley are you now only proposing labels be changed?

**Comment by avsculley** [comment]
@johnbeve @neilotte  The initial proposal was to 

1. change the label of 'planned act' to 'directed act';
2. change the alt label of 'planned/directed act' to 'intended act', which more accurately reflects the centrality of some prescription to the model (if you are familiar with the difference between intended and intentional acts, then hopefully I don't need to say more about that, but I can if requested);
3. change the def of 'act of planning' to reflect that its genus is now 'directed act'.

In addition to the original proposal:

4. changing the defs of subclasses of directed acts to reflect their new genera;
5. we should consider whether a 'planned act' class is needed, and whether any subclasses of 'directed act' are actually better categorized as planned acts. But, presumably, since planned acts are directed acts, 5 is probably not super pressing.

**Comment by mark-jensen** [comment]
> change the label of 'planned act' to 'directed act'

I do not like 'directed' as it suggests an order. How about 'Prescribed Act'?

> change the alt label of 'planned/directed act' to 'intended act'

"intentional act" will remain an alternate label for legacy reasons. Feel free to add "intended act" or "directed act" as extra alt labels. Maybe also keep "planned act" as an alt as well. It'll help users find it if they're searching for 'planned act'.

You are putting a lot into suspect terms like 'planned', 'intended', 'intentional', 'directed'. IMO this opens up a can of worms, but you seem to have a lot to say about it. Which is great! Please start a discussion about how to make better CCO's representation of  Intentional/Planned/Prescribed/Directed Acts.

**Comment by oliviahobai** [comment]
Going to recommend further discussion about the changes called for, then a more targeted issue and PR to develop. Cheers
